rcl_action/src/rcl_action/names.c

```diff
@@ -30,6 +32,10 @@ rcl_action_get_goal_service_name(
 {
   RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "allocator is invalid", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(action_name, RCL_RET_INVALID_ARGUMENT);
+  if (0 == strlen(action_name)) {
+    RCL_SET_ERROR_MSG("invalid empty action name");
+    return RCL_RET_INVALID_ARGUMENT;
```

Comment:
Suggest returning `RCL_RET_ACTION_NAME_INVALID`. Then it makes it possible for the action client/server to know if the provided name is empty (invalid) vs a null argument. Same for other functions and also update the docs to include the new return type.

Reply:
Alright, makes sense. Done in 276aed1dffbb375fa5bb8cde52ddc28fe6e7bc56.